### PR TITLE
Prompt User When Migrating Data from Rook to OpenEBS

### PR DESCRIPTION
**Pull request by rrpolanco**
#### What this PR does / why we need it:

Prompt user when data migration path from Rook to OpenEBS is detected.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: addons/openebs/3.3.0/install.sh [hunk 1]
```diff
@@ -40,17 +40,45 @@ function openebs() {
...
+    printf "    As part of this, all pods mounting PVCs will be stopped, taking down the application.\n"
+    printf "\n"
+    printf "    Copying the data currently stored within rook-ceph will require at least %s of free space across the cluster.\n" "$ceph_disk_usage_total"
+    printf "    It is recommended to take a snapshot or otherwise back up your data before proceeding.\n${NC}"
+    printf "\n"
+    printf "Would you like to continue? "
+
+    if ! confirmN; then
+        printf "Not migrating\n"
+        bail
```
Comment (emosbaugh) [inline addons/openebs/3.3.0/install.sh:77]:
```suggestion
        bail "Not migrating"
```

File: addons/openebs/template/base/install.sh [hunk 2]
```diff
@@ -45,12 +45,40 @@ function openebs() {
 function openebs_maybe_migrate_from_rook() {
     if [ -z "$ROOK_VERSION" ]; then
         if kubectl get ns | grep -q rook-ceph; then
+            prompt_migrate_from_rook
```
Comment (emosbaugh) [inline addons/openebs/template/base/install.sh:48]: can we move this to `openebs_pre_init`? you will have to check that kubernetes is installed.

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/openebs/template/base/install.sh [hunk 3]
```diff
@@ -45,12 +45,40 @@ function openebs() {
...
             rook_ceph_to_sc_migration "$OPENEBS_LOCALPV_STORAGE_CLASS"
             DID_MIGRATE_ROOK_PVCS=1 # used to automatically delete rook-ceph if object store data was also migrated
         fi
     fi
 }
 
+function prompt_migrate_from_rook() {
+    local ceph_disk_usage_total
+    local rook_ceph_exec_deploy
```
Comment (emosbaugh) [inline addons/openebs/template/base/install.sh:57]:
```suggestion
    local rook_ceph_exec_deploy=rook-ceph-operator
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/openebs/3.3.0/install.sh [hunk 4]
```diff
@@ -40,17 +40,45 @@ function openebs() {
...
 }
 
+function prompt_migrate_from_rook() {
+    local ceph_disk_usage_total
+    local rook_ceph_exec_deploy
+
+    if kubectl get deployment -n rook-ceph rook-ceph-tools &>/dev/null; then
+        rook_ceph_exec_deploy=rook-ceph-tools
+    fi
+    ceph_disk_usage_total=$(kubectl exec -n rook-ceph deployment/$rook_ceph_exec_deploy -- ceph df | grep TOTAL | awk '{ print $8$9 }')
```
Comment (emosbaugh) [inline addons/openebs/3.3.0/install.sh:62]: i think shellcheck will complain here

```suggestion
    ceph_disk_usage_total=$(kubectl exec -n rook-ceph "deployment/$rook_ceph_exec_deploy" -- ceph df | grep TOTAL | awk '{ print $8$9 }')
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/openebs/3.3.0/install.sh (hunk 4 above)
Comment (rrpolanco) [reply addons/openebs/3.3.0/install.sh:62]: actually, `shellcheck` did not flag this as an issue

**Review by emosbaugh — APPROVED** [review]

**Comment by rrpolanco** [comment]
> Testgrid Run(s) Executing @ https://testgrid.kurl.sh/run/pr-3638-4d6a15e-openebs-3.3.0-k8s-docker-localpv-2022-10-28T14:08:26Z

Failing test is **localpv upgrade from 2.6.0** is due to Minio add-on error:
```
⚙  Addon minio 2020-01-25T02-50-51Z
2022-10-28 14:25:22+00:00 namespace/minio created
2022-10-28 14:25:22+00:00 secret/minio-credentials created
2022-10-28 14:25:22+00:00 service/minio created
Error from server (InternalError): error when creating "./kustomize/minio/": Internal error occurred: failed calling webhook "admission-webhook.openebs.io": Post "https://admission-server-svc.openebs.svc:443/validate?timeout=5s": dial tcp 10.96.2.22:443: connect: connection refused
2022-10-28 14:25:22+00:00 deployment.apps/minio created
2022-10-28 14:25:22+00:00 An error occurred on line 35
+ KURL_EXIT_STATUS=1
```
This seems like a flak as other test runs pass.

**Pull request merged**